error: compile with VS2015 win32

Hi

I use cmake **3.7** and vs 2015 (Microsoft (R) C/C++ Optimizing Compiler Version **19.00.24210** for x86) to compile. My Eigen version is Eigen **3.2.8**. But I get some errors as following. Any suggestion about this?

Thanks a lot.
Yanming


sophus\test\core\tests.hpp(112): error C2672: 'Sophus::details::metric': no matching overloaded function found
sophus\test\core\tests.hpp(109): note: while compiling class template member function 'bool Sophus::LieGroupTests<Sophus::RxSO3<double,0>>::veeHatTest(void)'
sophus\test\core\tests.hpp(134): note: see reference to function template instantiation 'bool Sophus::LieGroupTests<Sophus::RxSO3<double,0>>::veeHatTest(void)' being compiled
Sophus\test\core\test_rxso3.cpp(70): note: see reference to class template instantiation 'Sophus::LieGroupTests<Sophus::RxSO3<double,0>>' being compiled
Sophus\test\core\test_rxso3.cpp(69): note: while compiling class template member function 'bool Sophus::Tests<double>::testLieProperties(void)'
Sophus\test\core\test_rxso3.cpp(61): note: see reference to function template instantiation 'bool Sophus::Tests<double>::testLieProperties(void)' being compiled
Sophus\test\core\test_rxso3.cpp(233): note: see reference to class template instantiation 'Sophus::Tests<double>' being compiled
sophus\test\core\tests.hpp(112): error C2782: 'unknown-type Sophus::details::metric(const T &,const T &)': template parameter 'T' is ambiguous
Sophus\sophus/test_macros.hpp(28): note: see declaration of 'Sophus::details::metric'
sophus\test\core\tests.hpp(112): note: could be 'Eigen::Matrix<double,4,1,0,4,1>'
sophus\test\core\tests.hpp(112): note: or       'Eigen::internal::workaround_msvc_stl_support<T>'
          with
          [
              T=Eigen::Matrix<double,4,1,0,4,1>
          ]
sophus\test\core\tests.hpp(112): error C2784: 'unknown-type Sophus::details::metric(const T &,const T &)': could not deduce template argument for 'const T &' from 'Eigen::Matrix<double,4,1,0,4,1>'
Sophus\sophus/test_macros.hpp(28): note: see declaration of 'Sophus::details::metric'
sophus\test\core\tests.hpp(114): error C3536: 'nrm': cannot be used before it is initialized

--- a/test/core/tests.hpp
+++ b/test/core/tests.hpp
@@ -109,7 +109,7 @@ class LieGroupTests {
   bool veeHatTest() {
     bool passed = true;
     for (size_t i = 0; i < tangent_vec_.size(); ++i) {
-      SOPHUS_TEST_APPROX(passed, tangent_vec_[i],
+      SOPHUS_TEST_APPROX(passed, Tangent(tangent_vec_[i]),
                          LieGroup::vee(LieGroup::hat(tangent_vec_[i])),
                          kSmallEps, "Hat-vee case: %", i);
     }

